torch/onnx/_internal/fx/_pass.py

```diff
@@ -216,11 +216,11 @@ def __init__(
         self,
         diagnostic_context: diagnostics.DiagnosticContext,
         module: torch.fx.GraphModule,
+        onnxfunction_dispatcher: onnxfunction_dispatcher.OnnxFunctionDispatcher,
     ):
         self.diagnostic_context = diagnostic_context
         self.module = module
-        self.registry = registration.OnnxRegistry()
-        self.dispatcher = function_dispatcher.OnnxDispatcher(self.registry)
+        self.onnxfunction_dispatcher = onnxfunction_dispatcher
```

Comment:
This should not be on generic `Analysis`. Dispatcher should be moved to the specific AnalysisPass that needs dispatcher. But I see this is a mistake in previous PR. So I won't block on this.

Reply:
That was part of the bug I mentioned this PR fixes. It makes no sense to spawn a new `OnnxRegistry` and `OnnxDispacther`. In a scenario with custom ops registered by users and stored at `ResolvedExportOptions.onnx_registry`. By creating a new one just for the `Analysis` class, the custom ops would be missed, resulting in a potential bug.

On a side note, `Analysis` and `Transform` classes were merged without discussion. `Analysis` doesn't have docstring to provide any insight on how it differs or interact with `Transform` and what it should do. Not sure if this is part of `Diagnostics` only or what, but I would like to know when such big changes are proposed.